test/sanity/code-smell/no-assert.py

```diff
@@ -5,7 +5,7 @@
 import re
 import sys
 
-ASSERT_RE = re.compile(r'.*(?<![-:a-zA-Z#][ -])\bassert\b(?!:).*')
+ASSERT_RE = re.compile(r'^\s*assert[^a-z0-9_:]')
```

Comment:
won't that allow for a expression like `stuff or assert ...` ?

Reply:
Yes, it will. Thankfully that isn't valid Python syntax.

Reply:
ah, TIL